main.tf

```
@@ -249,8 +249,8 @@ data "aws_iam_policy_document" "default" {
     actions = distinct(compact(var.iam_actions))
 
     resources = [
-      join("", aws_elasticsearch_domain.default.*.arn),
-      "${join("", aws_elasticsearch_domain.default.*.arn)}/*"
+      join("", aws_elasticsearch_domain.default[*].arn),
```

For future reference, we want to use `one()` instead of `join` where `one()` is applicable. 

```suggestion
      one(aws_elasticsearch_domain.default[*].arn),
```

---

The 'one' function transforms a list with either zero or one elements into either a null value or the value of the first element.

This join, for empty list, will return "" instead of null. 

This difference did not allow me to migrate to one.

---

The majority of places I replaced with one did not work because we got used to "" if the argument is absent